xmlhttprequest.php

```diff
@@ -43,6 +43,8 @@
 	log_event( LOG_AJAX, 'DEPRECATED: Calling {' . $t_function . '}. Use REST API instead.' );
 	call_user_func( $t_function );
 } else {
-	log_event( LOG_AJAX, 'Unknown function for entry point = ' . $t_function );
-	echo 'unknown entry point';
+	$t_msg = 'Unknown function for entry point: ' . $t_function;
+	log_event( LOG_AJAX, $t_msg );
+	header( 'HTTP/1.1 500 ' . $t_msg, false, 500 );
+	echo $t_msg;
```

Comment:
I don't think XSS applies to HTTP headers (no HTML or Javascript executed there), but I'll check if header injection is possible (although offhand I don't think PHP allows this).